crates/pathfinder/src/state/sync.rs

```diff
@@ -558,3 +571,390 @@ fn deploy_contract(
+            &self,
+            _: BlockNumberOrTag,
+        ) -> Result<reply::StateUpdate, SequencerError> {
+            unimplemented!()
+        }
+
+        async fn eth_contract_addresses(
+            &self,
+        ) -> Result<reply::EthContractAddresses, SequencerError> {
+            unimplemented!()
+        }
+    }
+
+    async fn l1_noop(
+        _: mpsc::Sender<l1::Event>,
+        _: Web3<FakeTransport>,
+        _: ethereum::Chain,
+        _: Option<ethereum::log::StateUpdateLog>,
+    ) -> anyhow::Result<()> {
+        Ok(())
+    }
+
+    async fn l2_noop(
+        _: mpsc::Sender<l2::Event>,
+        _: impl sequencer::ClientApi,
+        _: Option<(StarknetBlockNumber, StarknetBlockHash)>,
+    ) -> anyhow::Result<()> {
+        Ok(())
+    }
+
+    lazy_static::lazy_static! {
+        static ref A: StarkHash = StarkHash::from_be_slice(&[0xA]).unwrap();
+        static ref B: StarkHash = StarkHash::from_be_slice(&[0xB]).unwrap();
+        static ref ETH_ORIG: ethereum::EthOrigin = ethereum::EthOrigin {
+            block: ethereum::BlockOrigin {
+                hash: EthereumBlockHash(H256::zero()),
+                number: EthereumBlockNumber(0),
+            },
+            log_index: EthereumLogIndex(0),
+            transaction: ethereum::TransactionOrigin {
+                hash: EthereumTransactionHash(H256::zero()),
+                index: EthereumTransactionIndex(0),
+            },
+        };
+        pub static ref STATE_UPDATE_LOG0: ethereum::log::StateUpdateLog = ethereum::log::StateUpdateLog {
+            block_number: StarknetBlockNumber(0),
+            global_root: GlobalRoot(*A),
+            origin: ETH_ORIG.clone(),
+        };
+        pub static ref STATE_UPDATE_LOG1: ethereum::log::StateUpdateLog = ethereum::log::StateUpdateLog {
+            block_number: StarknetBlockNumber(1),
+            global_root: GlobalRoot(*B),
+            origin: ETH_ORIG.clone(),
+        };
+        pub static ref BLOCK0: reply::Block = reply::Block {
+            block_hash: Some(StarknetBlockHash(*A)),
+            block_number: Some(StarknetBlockNumber(0)),
+            parent_block_hash: StarknetBlockHash(StarkHash::ZERO),
+            state_root: Some(GlobalRoot(*A)),
+            status: reply::Status::AcceptedOnL1,
+            timestamp: crate::core::StarknetBlockTimestamp(0),
+            transaction_receipts: vec![],
+            transactions: vec![],
+        };
+        pub static ref BLOCK1: reply::Block = reply::Block {
+            block_hash: Some(StarknetBlockHash(*B)),
+            block_number: Some(StarknetBlockNumber(1)),
+            parent_block_hash: StarknetBlockHash(*A),
+            state_root: Some(GlobalRoot(*B)),
+            status: reply::Status::AcceptedOnL2,
+            timestamp: crate::core::StarknetBlockTimestamp(1),
+            transaction_receipts: vec![],
+            transactions: vec![],
+        };
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
```

Comment:
Why  do we need `multi_thread` here?

Reply:
Yeah, that's because of `tokio::task::block_in_place` usage, otherwise you get `thread 'state::sync::tests::yout_test_here' panicked at 'can call blocking only when running on the multi-threaded runtime'` :smile: .